Prox/ProxTests/OpenHoursTests.swift

```diff
@@ -66,91 +75,71 @@ class OpenHoursTests: XCTestCase {
-
-        date.day = 8
-        XCTAssertTrue(hours.isOpen(atTime: date.date!), "Expected open for \(date)")
-
-        date.hour = 2
-        date.minute = 0
-        XCTAssertFalse(hours.isOpen(atTime: date.date!), "Expected closed for \(date)")
+        // current time 7th November 2016 06:59
+        var date = dateComponents(withYear: 2016, month: 11, day: 7, hour: 6, minute: 59).date!
+
+        // open time 07:00, closing time 21:00
+        var hours = OpenHours(hours: [.monday : (openTime: dateComponents(withHour: 7, minute: 0), closeTime: dateComponents(withHour: 21, minute: 0))])
+
+        XCTAssertFalse(hours.isOpen(atTime: date), "Expected closed for \(date)")
+
+        // current time 7th November 2016 07:00
+        date = dateComponents(withYear: 2016, month: 11, day: 7, hour: 7, minute: 0).date!
+        XCTAssertTrue(hours.isOpen(atTime: date), "Expected open for \(date)")
+
+        // current time 7th November 2016 21:00
+        date = dateComponents(withYear: 2016, month: 11, day: 7, hour: 21, minute: 0).date!
+        XCTAssertFalse(hours.isOpen(atTime: date), "Expected closed for \(date)")
+
+        // Monday open time 07:00, closing time 00:00
+        // Tuesday open time 07:00, closing time 02:00
+        let opening = dateComponents(withHour: 7, minute: 0)
+        hours = OpenHours(hours: [.monday : (openTime: opening, closeTime: dateComponents(withHour: 0, minute: 0)),
+                                  .tuesday : (openTime: opening, closeTime: dateComponents(withHour: 2, minute: 0)),
+                                  .wednesday : (openTime: opening, closeTime: dateComponents(withHour: 2, minute: 0))])
+
+        // current time 7th November 2016 23:59
+        // open time 7th November 2016 07:00
+        // close time 8th November 2016 00:00
+        date = dateComponents(withYear: 2016, month: 11, day: 7, hour: 23, minute: 59).date!
+        XCTAssertTrue(hours.isOpen(atTime: date), "Expected open for \(date)")
+
+        // current time 7th November 2016 00:00
+        // open time 7th November 2016 07:00
+        // close time 8th November 2016 00:00
+        date = dateComponents(withYear: 2016, month: 11, day: 7, hour: 0, minute: 0).date!
+        XCTAssertFalse(hours.isOpen(atTime: date), "Expected closed for \(date)")
+
+        // current time 9th November 2016 00:00
+        // open time 8th November 2016 07:00
+        // close time 9th November 2016 02:00
+        date = dateComponents(withYear: 2016, month: 11, day: 9, hour: 0, minute: 0).date!
+        XCTAssertTrue(hours.isOpen(atTime: date), "Expected open for \(date)")
+
+        // current time 9th November 2016 02:00
+        // open time 8th November 2016 07:00
+        // close time 9th November 2016 02:00
+        date = dateComponents(withYear: 2016, month: 11, day: 9, hour: 2, minute: 0).date!
+        XCTAssertFalse(hours.isOpen(atTime: date), "Expected closed for \(date)")
```

Comment:
I'm pretty sure that we have that test in there already, but I'll check. The fact that you can't see it means that I do need to make the tests clearer